common/src/main/java/com/wynntils/overlays/StatusEffectsOverlay.java

```diff
@@ -108,4 +133,38 @@ private void updateTextRenderSetting() {
     protected TextRenderSetting getTextRenderSetting() {
         return textRenderSetting;
     }
+
+    private Stream<StatusEffectWithProperties> stackEffects(List<StatusEffect> effects) {
+        LinkedHashMap<String, StatusEffectWithProperties> effectsToRender = new LinkedHashMap<>();
+
+        for (var effect : effects) {
+            StatusEffectWithProperties entry =
+                    effectsToRender.get(effect.asString().getString());
+            if (entry == null) {
+                entry = new StatusEffectWithProperties(effect);
+                effectsToRender.put(effect.asString().getString(), entry);
+            }
+            entry.count += 1;
+        }
+        return effectsToRender.values().stream();
+    }
+
+    private StyledText getTextToRender(StatusEffectWithProperties effect) {
+        return effect.count > 1
+                ? StyledText.fromString(effect.count + "x ").append(effect.asString())
+                : effect.asString();
+    }
+
+    private static final class StatusEffectWithProperties extends StatusEffect {
+        public int count = 0;
+
+        private StatusEffectWithProperties(
+                StyledText name, StyledText modifier, StyledText displayedTime, StyledText prefix) {
+            super(name, modifier, displayedTime, prefix);
+        }
+
+        private StatusEffectWithProperties(StatusEffect effect) {
+            this(effect.getName(), effect.getModifier(), effect.getDisplayedTime(), effect.getPrefix());
+        }
+    }
```

Comment:
Personally, I think this class has little use and you could just use `Pair<Integer, StatusEffect>` (since it is not used in many places).

You don't necessary have to do this, but I think it would be a bit clearer.

Reply:
I have some reasons for not using the `Pair` class:

- This way it is easier to add more options later, like stacking effects with the same name but a different modifier
-  The `Pair` class contains immutable data, which doesn't fit this use case because the integer needs to be incremented. It would require either creating a new Pair on each iteration or creating an `ImmutablePair` type.

Reply:
Okay then. Let's rename this class to something easier to read. Perhaps RenderedStatusEffect.

Secondly, what do you think of not extending status effect, but rather having it as an immutable field? That is closer to what I want / what we have done before.

Lastly, please make the count field private. Fields are private. If needed, use a getter.

Reply:
Also, `getTextToRender` could easily be moved here as a member method.

Reply:
And as for the `getTextToRender` format. Can you make it so it's something like "+2x40% Attack Speed"?

Reply:
Keep it here for now.